### PR TITLE
Add typings for codemirror's lint addon

### PR DESCRIPTION
**Pull request by CalvinFernandez**

### CODE REVIEW

File: codemirror/codemirror.d.ts [hunk 1]
```diff
@@ -1004,6 +1007,41 @@ declare module CodeMirror {
...
+        async: boolean;
+        getAnnotations: AnnotationsCallback;
+        hasGutters: boolean;
+    }
+
+    /**
+     * A function that calls the updateLintingCallback with any errors found during the linting process.
+     */
+    interface AnnotationsCallback {
+        (content: string, updateLintingCallback: UpdateLintingCallback, options: any, codeMirror: Editor): void;
```
Comment (thasner) [inline codemirror/codemirror.d.ts]: can `options` be typed?

File: codemirror/codemirror.d.ts [hunk 2]
```diff
@@ -1004,6 +1007,41 @@ declare module CodeMirror {
...
+    }
+
+    /**
+     * A function that, given an array of annotations, updates the CodeMirror linting GUI with those annotations
+     */
+    interface UpdateLintingCallback {
+        (codeMirror: any, annotations: Annotation[]): void;
```
Comment (thasner) [inline codemirror/codemirror.d.ts]: `codeMirror` argument is `any` ??

**Comment by vvakame** [comment]
*codemirror/codemirror.d.ts*

to author(@mihailik). could you review this PR?
:+1: or :-1:?

check list

* [ ] pass the Travic-CI test?

**Comment by CalvinFernandez** [comment]
Bump.

**Comment by thasner** [comment]
:+1:

**Comment by mihailik** [comment]
:+1:

**Comment by vvakame** [comment]
@mihailik @thasner thanks!
@CalvinFernandez thanks mate!

**Pull request merged**